Readme.md

```diff
@@ -0,0 +1,494 @@
+
+For this exercise, we would use the document we created earlier with create operation `{restaurant_id: "restaurantId"}` and update the
+`restaurant_id` with a more realistic value.
+
+First, we need to query the document using `Filters`, similar to Read operation earlier.
+
+```kotlin
+val collection = db.getCollection<Restaurant>("restaurants")
+val queryParam = Filters.eq("restaurant_id", "restaurantId")
+```
+
+Then we can set the `restaurant_id` with a random integer value using `Updates`.
+
+```kotlin
+val updateParams = Updates.set("restaurant_id", Random.nextInt().toString())
+```
+
+And finally, we use `updateOne` to update the document.
+
+```kotlin
+collection.updateOne(filter = queryParam, update = updateParams).also {
+    println("Total docs modified ${it.matchedCount} and fields modified ${it.modifiedCount}")
+}
+```
+
+In the above example, we were already aware of which document we want to update, restaurant with an id `restauratantId` but in the real-world app
+that might not be the case. So then you would first look for the document and then update it (two operations for one task) which can lead to
+performance issues or race conditions.
+
+To overcome this, we could use `findOneAndUpdate` that allow you to combine both in an atomic operation.
+
+Another variation of the same could be updating multiple documents with one call and `updateMany` come in handy for such use cases. In our CRUD
+app, we would update `cuisine` all restaurants to your favourite type and `borough` to Brooklyn.
+
+```kotlin
+suspend fun updateMultipleDocuments(db: MongoDatabase) {
+    val collection = db.getCollection<Restaurant>("restaurants")
+    val queryParam = Filters.eq(Restaurant::cuisine.name, "Chinese")
+    val updateParams = Updates.combine(
+        Updates.set(Restaurant::cuisine.name, "Indian"),
+        Updates.set(Restaurant::borough.name, "Brooklyn")
+    )
+
+    collection.updateMany(filter = queryParam, update = updateParams).also {
+        println("Total docs matched ${it.matchedCount} and modified ${it.modifiedCount}")
+    }
+}
+```
+
+In these examples, we used `set` and `combine` with `Updates` but there is much more to explore that allows us to do a more intuitive operation
+like set currentDate or timestamp, increase or decrease the value of the field, etc. You can refer to the
+[docs](https://mongodb.github.io/mongo-java-driver/4.9/apidocs/mongodb-driver-core/com/mongodb/client/model/Updates.html) for the complete list.
+
+---------------------
+
+## Delete
+
+Now let's see the last operation of our CRUD app i.e. Delete and start with how to delete a single document and here we would be
+using `findOneAndDelete` instead `deleteOne` which also returns the deleted document as output.
+
+```kotlin
+val collection = db.getCollection<Restaurant>(collectionName = "restaurants")
+val queryParams = Filters.eq("restaurant_id", "restaurantId")
+
+collection.findOneAndDelete(filter = queryParams).also {
```

Comment:
I would not prioritize `findOneAndDelete` over `deleteOne` in your example code. For a lot of people, they likely won't use `findOneAndUpdate` or `deleteOneAndUpdate`. Maybe show both ways if you really want to keep them in.

Reply:
I would prefer to keep it.